src/jit/importer.cpp

```diff
@@ -3607,7 +3614,33 @@ GenTreePtr Compiler::impIntrinsic(CORINFO_CLASS_HANDLE  clsHnd,
             retNode = op1;
             break;
 #endif
-
+        // Implement ByReference Ctor.  This wraps the assignment of the ref to the span field.
+        // In effect this is just a substitution.  The parameter byref will be substituted for the
+        // call node.
+        case CORINFO_INTRINSIC_ByReference_Ctor:
+        {
+            // Remove call to constructor and directly assign the byref passed
+            // to the call to the first slot of the ByReference struct.
```

Comment:
The intrinsic is for the ByReference struct itself.  Any other struct that includes the ByReference type as a field will be handled the same as span. (assuming the runtime is identifying it as one of the special ByRefLike types)  This is a little different from the old implementation you used before.

`
struct NewSpan {
    int hash;
    ByReference _pointer;
    int length;
}
`
Would still work assuming that the change happened in the runtime to recognize this as a ByRefLike value type.  (I'm assuming that we could add an attribute for this if we start making lots more of these)